File: 0701/SeoyeonHong/신규 아이디 추천.py
# https://school.programmers.co.kr/learn/courses/30/lessons/72410
def solution(new_id):
    answer = ''
    for char in new_id.lower():
        if 96 < ord(char) < 123: # 알파벳 소문자일 경우
            answer += char
        elif char.isdigit() or char == '-' or char == '_':
            answer += char
        elif char == '.':
            if answer and answer[-1] != '.':
                answer += char
        
    if answer == '': # new_id가 빈 문자열일 경우 a 대입
        answer = 'a'
    elif len(answer) >= 16:
        answer = answer[:15]

    if answer[-1] == '.':
        answer = answer[:-1]
            
    if len(answer) <= 2:
        while len(answer) < 3:
            answer += answer[-1]
                            
    return answer

File: 0701/SeoyeonHong/신규 아이디 추천.java
// https://school.programmers.co.kr/learn/courses/30/lessons/72410?language=java

class Solution {
    public String solution(String new_id) {
        String answer = "";
        new_id = new_id.toLowerCase();
        for(int i = 0; i < new_id.length(); i++){
            char c = new_id.charAt(i);
            if(Character.isAlphabetic(c) || Character.isDigit(c) || c == '-' || c == '_'){         
                answer += c; 
            } 
            else if(c == '.'){
                if(!answer.isEmpty() && answer.charAt(answer.length()-1) != '.' ){
                    answer += c;
                }
            }
        }

        if(answer.length() > 0 && answer.endsWith(".")){
            answer = answer.substring(0, answer.length()-1);
        }
        
        if(answer.isEmpty()){
            answer = "a";
        }
        if(answer.length() >= 16){
            answer = answer.substring(0, 15);
            if(answer.endsWith(".")){
                answer = answer.substring(0, answer.length()-1);
            }
        }
        if(answer.length() <= 2){
            while(answer.length() < 3){
                answer += answer.charAt(answer.length()-1);
            }
        }
        return answer;
    }
}